### PR TITLE
Makman2/linterups

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

File: Developers/Writing_Linter_Bears.rst [hunk 1]
```diff
@@ -205,6 +205,10 @@ We can test our bear like this
     by their **filename** and display them based on their
     **classname**.
 
+Normally, providing a severity-map is not needed, as *coala* has a default
```
Comment (Adrianzatreanu) [inline Developers/Writing_Linter_Bears.rst]: nooo no more italic on coala, i have taken thoughts and we dont want that :D

Comment (Makman2) [inline Developers/Writing_Linter_Bears.rst]: haha alright :D

**Comment by Adrianzatreanu** [comment]
ack c0336f81c9b54f87ebecaf151e7e2b5f81850a9a

File: Developers/Writing_Linter_Bears.rst [hunk 2]
```diff
@@ -71,6 +71,17 @@ You can use these parameters to construct the command line arguments. The
 linter expects from you to return an argument sequence here. A tuple is
 preferred. We will do this soon for ``PylintTutorialBear``.
 
+.. note::
+
+    ``create_arguments`` can also be a non-staticmethod. In this case you also
```
Comment (Adrianzatreanu) [inline Developers/Writing_Linter_Bears.rst]: isnt it non-static method as in two words?

File: Developers/Writing_Linter_Bears.rst [hunk 3]
```diff
@@ -71,6 +71,17 @@ You can use these parameters to construct the command line arguments. The
...
+    need to prepend ``self`` to the parameters in the signature. Some functions
```
Comment (Adrianzatreanu) [inline Developers/Writing_Linter_Bears.rst]: what is ``prepend``?

File: Developers/Writing_Linter_Bears.rst [hunk 4]
```diff
@@ -33,7 +33,7 @@ Writing the Bear
 ----------------
```
Comment (Adrianzatreanu) [inline Developers/Writing_Linter_Bears.rst]: the commit message should explain an action: ``Remove comma before "that"``

**Comment by Adrianzatreanu** [comment]
ack f1aa603921f484fdb521db648f836194d8f10ce2

File: Developers/Linter_Bears_Advanced.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file):
+            config_file = ("value1 = 1\n"
+                           "value=2 = 2")
+            return config_file
+
+The string returned from this method is written into a temporary file before
```
Comment (Adrianzatreanu) [inline Developers/Linter_Bears_Advanced.rst]: "returned by"

**Comment by Adrianzatreanu** [comment]
ack 88260f0629599731a53a9f4a29649c2b38bce60b

**Comment by Adrianzatreanu** [comment]
ack a7c0789d1d37bc6d83cc77c7c7d22f4c23b1902d

File: Developers/Writing_Linter_Bears.rst (hunk 2 above)
Comment (Makman2) [inline Developers/Writing_Linter_Bears.rst]: I wanted to refer to the `@staticmethod` decorator. Though yes you are right, this is no good english :3
I'll rephrase to
> ``create_arguments`` doesn't have to be a static method.

File: Developers/Writing_Linter_Bears.rst (hunk 3 above)
Comment (Makman2) [inline Developers/Writing_Linter_Bears.rst]: the opposite of append, putting sth at the beginning of sth :3

File: Developers/Writing_Linter_Bears.rst (hunk 4 above)
Comment (Makman2) [inline Developers/Writing_Linter_Bears.rst]: does not fit, seems I need to cut down the tag...

**Comment by Makman2** [comment]
reack
a68d618 Writing_Linter_Bears.rst: Wrap text
7636234 Writing_Linter_Bears.rst: Remove last sentence
95a6b1d Self_Correcting_Bears.rst: Delete
29163fc Linter_Bears.rst: Refer to API for severities

**Comment by sils1297** [comment]
ack https://github.com/coala/documentation/pull/68/commits/d2a8df26a4ccf6e564e4bbd492c8f7ace1a0de1a

**Comment by sils1297** [comment]
ack https://github.com/coala/documentation/pull/68/commits/c6d9cee12aeae701ef8fcb8beb98a8bda80b86a5 https://github.com/coala/documentation/pull/68/commits/d89b0eef6061d4c5d641dd5354cf268aebc5e87d

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**